src/EditorFeatures/Core/Implementation/IntelliSense/Completion/DescriptionModifyingCompletionItem.cs

```diff
@@ -53,6 +54,24 @@ public override CompletionListProvider CompletionProvider
             }
         }
 
+        /// <summary>
+        /// Returns a string that should be added to the description of the given completion item
+        /// if a snippet exists with a shortcut that matches the completion item's insertion text.
+        /// </summary>
+        private Task<string> GetSnippetExpansionNoteForCompletionItemAsync(CompletionItem completionItem)
+        {
+            var completionRules = _completionService.GetCompletionRules();
+            var insertionText = completionRules.GetTextChange(completionItem, '\t').NewText;
```

Comment:
When we decide whether to add the "tab twice to insert the snippet" text, we make that decision based on the insertion text. This is basically a hack that makes the CompletionRules pretend the item was committed on tab or double click, to avoid any special casing they might do for committing on open paren or similar.